src/learn/internal/learnservice.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "learnservice.h"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QBuffer>
+
+#include "log.h"
+
+using namespace mu::learn;
+using namespace mu::network;
+using namespace mu::async;
+
+LearnService::LearnService(QObject* parent)
+    : QObject(parent)
+{
+}
+
+Promise<Playlist> LearnService::startedPlaylist() const
+{
+    return Promise<Playlist>([this](Promise<Playlist>::Resolve resolve,
+                                    Promise<Playlist>::Reject) {
+        QUrl playlistUrl = configuration()->startedPlaylistUrl();
+        if (playlistUrl.isEmpty()) {
+            return resolve({});
+        }
+
+        resolve(requestPlaylist(playlistUrl));
```

Comment:
It's not clear why a promise is used here if all the code is synchronous. Or is it for the future? to be able to use the network manager's asynchronous mode.
Maybe you should use the asynchronous mode right now? Or execute this code in a separate thread